statsmodels/graphics/functional.py

```diff
@@ -437,26 +441,25 @@ def _band_quantiles(band):
             ax.plot(xdata, outlier,
                     ls='--', alpha=0.7, label=label)
 
-    if labels is not None:
-        handles, labels = ax.get_legend_handles_labels()
-
-        # Proxy artist for fill_between legend entry
-        # See http://matplotlib.org/1.3.1/users/legend_guide.html
-        plt = _import_mpl()
-        for label, fill_between in zip(['50% HDR', '90% HDR'], fill_betweens):
-            p = plt.Rectangle((0, 0), 1, 1,
-                              fc=fill_between.get_facecolor()[0])
-            handles.append(p)
-            labels.append(label)
-
-        by_label = OrderedDict(zip(labels, handles))
-        if len(outliers) != 0:
-            by_label.pop('dataset')
-            by_label.pop('Median')
-            by_label.pop('50% HDR')
-            by_label.pop('90% HDR')
-
-        ax.legend(by_label.values(), by_label.keys(), loc='best')
+    handles, labels = ax.get_legend_handles_labels()
+
+    # Proxy artist for fill_between legend entry
+    # See http://matplotlib.org/1.3.1/users/legend_guide.html
+    plt = _import_mpl()
+    for label, fill_between in zip(['50% HDR', '90% HDR'], fill_betweens):
+        p = plt.Rectangle((0, 0), 1, 1,
+                          fc=fill_between.get_facecolor()[0])
+        handles.append(p)
+        labels.append(label)
+
+    by_label = OrderedDict(zip(labels, handles))
+    if len(outliers) != 0:
+        by_label.pop('dataset')
```

Comment:
This is now either redundant or will raise

Reply:
oh I missed this one!